doc/dynamic-environments/configuration.mkd

```diff
@@ -167,9 +167,8 @@ This defaults to 'https://forgeapi.puppetlabs.com'
 
 #### authorization_token
 
-The 'authorization_token' setting allows you to provide a token for authenticating to a
-custom Forge server. When set, 'baseurl' must also be set.
-You will need to prepend your token with 'Bearer ' if using Artifactory as your Forge server.
+The 'authorization_token' setting allows you to provide a token for authenticating to a Forge server.
+You will need to prepend your token with 'Bearer ' to authenticate to the Forge or when using your own Artifactory server.
```

Comment:
```suggestion
You will need to prepend your token with 'Bearer ' to authenticate to the Forge or when using Artifactory as your Forge server.
```

Reply:
I suggest waiting to merge this until https://github.com/puppetlabs/forge-ruby/pull/89 is merged and released, since it obviates the need to include 'Bearer'

Reply:
Looks like that should work for Artifactory-based Forge too?

Reply:
I think so, although I haven't checked that manually - I did manually verify just now that the version of forge-ruby I just released (3.1.0) will allow this to work without prepending `Bearer ` in the r10k.yaml setting. Otherwise, the only thing stopping r10k from authenticating before was the check for a `baseurl` that's removed in this PR.

Unless I'm missing something, it seems like the options here are to:
* Update this README to suggest specifying `Bearer <your-key>` as the `authorization_token` setting for now (regardless of whether you're authenticating to the forge or to artifactory), and leave the forge-ruby dependency as-is
or
* Update the forge-ruby dependency to 3.1.0 and don't mention `Bearer` at all - the  biggest impact from this would probably be that the default forge API url set by forge-ruby would update from https://forgeapi.puppetlabs.com to https://forgeapi.puppet.com.

Reply:
I guess a third option would be to add new logic in r10k to prepend `Bearer ` to the key as needed - it wouldn't affect forge-ruby behavior in 3.1.0, and it would let people omit `Bearer ` in their r10k.yaml file if they're using an older forge-ruby version.

Reply:
We talked about that and decided we didn't know enough about what token requirements other custom Forge implementations might have, and didn't want to introduce magic that was going to bite us. Though I guess if the Forge gem is doing that now, that point is kinda moot.

Reply:
@caseywilliams Would the URL update in 3.1.0 that you mentioned cause breakages? I know we had some trouble updating that URL in r10k, so we're still using the `puppetlabs` one.

Reply:
Yeah, it might - if I remember correctly, the PE acceptance tests were breaking last time we tested using the updated URL in r10k. forgeapi.puppet.com is really the current canonical place to find the API, so everyone should switch over eventually, but the two biggest issues in updating would be (as far as I can tell) (1) if a user had allowlisted access in their network to forgeapi.puppetlabs.com and then suddenly lost access when the url updated but their rules had not, and (2) needing to schedule the acceptance test update work

Reply:
looks like bumping the puppet_forge gem to 3.1.0 will involve packaging a bunch of additional dependencies in pe-r10k-vanagon, since faraday 1.3+ has a bunch more dependencies than the version we're currently on.

I'm thinking it might be best to open a separate ticket/issue for bumping puppet_forge, removing the docs around `Bearer `, and updating the Forge API url. Would you mind doing that @caseywilliams?

Reply:
Sure, wasn't sure exactly which jira project was appropriate but I filed [PE-32770](https://tickets.puppetlabs.com/browse/PE-32770).